src/tests/unit/platform/test_alertsv7_api.py

```diff
@@ -547,25 +503,97 @@ def on_post(url, body, **kwargs):
     assert a.workflow_.status == "OPEN"
 
 
-def test_query_watchlistalert_facets(cbcsdk_mock):
-    """Test a watchlist alert facet query."""
+def test_query_intrusion_detection_system_with_all_bells_and_whistles(cbcsdk_mock):
+    """Test a IDS alert query with all options selected."""
 
     def on_post(url, body, **kwargs):
-        assert body == {"query": "Blort", "criteria": {'type': ['WATCHLIST'], "workflow_status": ["OPEN"]},
-                        "terms": {"rows": 0, "fields": ["REPUTATION", "STATUS"]}}
-        return {"results": [{"field": {},
-                             "values": [{"id": "reputation", "name": "reputationX", "total": 4}]},
-                            {"field": {},
-                             "values": [{"id": "status", "name": "statusX", "total": 9}]}]}
+        assert body == {"query": "Blort",
+                        "rows": 2,
+                        "criteria": {"device_id": ["6023"], "device_name": ["HAL"],
+                                     "device_os": ["LINUX"], "device_os_version": ["0.1.2"],
+                                     "device_username": ["JRN"],
+                                     "device_internal_ip": ["10.10.8.2"], "device_external_ip": ["10.10.10.55"],
+                                     "id": ["S0L0"],
+                                     "severity": ["6"], "device_policy_id": ["8675309"],
+                                     "device_policy": ["Strict"], "device_uem_id": ["uemId"],
+                                     "device_target_value": ["HIGH"], "threat_name": ["dangerous"],
+                                     "threat_id": ["B0RG"], "type": ["INTRUSION_DETECTION_SYSTEM"],
+                                     "policy_applied": ["APPLIED"],
+                                     "reason_code": ["ATTACK_VECTOR"], "run_state": ["RAN"], "sensor_action": ["DENY"],
+                                     "alert_notes_present": True, "attack_tactic": ["tactic"],
```

Comment:
lower case t for true in json

Reply:
this doesn't work for building the request body

Reply:
Yeah pythonic JSON uses True and None for true and null